storage/remote/queue_manager.go

```diff
@@ -756,7 +756,6 @@ func (s *shards) stop() {
 
 	// Force an unclean shutdown.
 	s.hardShutdown()
-	<-s.done
```

Comment:
Waiting for `s.done` to be closed is necessary. The `stop()` function needs to wait for all shards to shutdown before returning. Otherwise, after resharding it will be possible for newer samples to be sent out before older samples in the same series are flushed. Removing this line breaks the strict ordering guarantee of remote write.

Reply:
@csmarchbanks thanks for the pointers. I removed this line since I made this a buffered channel above. I am not experiencing any issue, just came across this as an improvement. I agree that its not a potential bug per se. The go channels, I believe is FIFO in nature so it will guarantee the ordering regardless. However, I agree that since its working as designed, we don't need this PR. Thanks again. Closing this for now.